src/Gaufrette/Adapter/AsyncAwsS3.php

```diff
@@ -0,0 +1,339 @@
+<?php
+
+namespace Gaufrette\Adapter;
+
+use AsyncAws\SimpleS3\SimpleS3Client;
+use Gaufrette\Adapter;
+use Gaufrette\Util;
+
+/**
+ * Amazon S3 adapter using the AsyncAws.
+ *
+ * @author  Michael Dowling <mtdowling@gmail.com>
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class AsyncAwsS3 implements Adapter, MetadataSupporter, ListKeysAware, SizeCalculator, MimeTypeProvider
+{
+    /** @var SimpleS3Client */
+    protected $service;
+    /** @var string */
+    protected $bucket;
+    /** @var array */
+    protected $options;
+    /** @var bool */
+    protected $bucketExists;
+    /** @var array */
+    protected $metadata = [];
+    /** @var bool */
+    protected $detectContentType;
+
+    /**
+     * @param SimpleS3Client $service
+     * @param string   $bucket
+     * @param array    $options
+     * @param bool     $detectContentType
+     */
+    public function __construct(SimpleS3Client $service, $bucket, array $options = [], $detectContentType = false)
+    {
+        if (!class_exists(SimpleS3Client::class)) {
```

Comment:
this check is useless. If the class does not exist, it is impossible to pass an argument to `$service` due to the typehint. So this is dead code.

Reply:
That is true. 

Thank you